onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -633,87 +707,80 @@ def list(self, request, *args, **kwargs):
 
-                if new_validation_status_uid is None:
-                    http_status = status.HTTP_400_BAD_REQUEST
-                    response = {"detail": _("No validation_status.uid provided")}
-                else:
-                    # Create new validation_status object
-                    new_validation_status = get_validation_status(
-                        new_validation_status_uid, xform, request.user.username)
-
-                    # 3 scenarios to update submissions
-
-                    # First scenario / Modify submissions based on user's query
-                    if payload.get("query"):
-                        # Validate if query is valid.
-                        try:
-                            query.update(payload.get("query"))
-                        except ValueError:
-                            raise ParseError(_("Invalid query: %(query)s"
-                                               % {'query': json.dumps(payload.get("query"))}))
-
-                        query_kwargs = {
-                            "query": json.dumps(query),
-                            "fields": '["_id"]'
-                        }
-
-                        cursor = ParsedInstance.query_mongo_no_paging(**query_kwargs)
-                        submissions_ids = [record.get("_id") for record in list(cursor)]
-                        filter_.update({"id__in": submissions_ids})
-
-                    # Second scenario / Modify submissions based on list of ids
-                    elif payload.get("submissions_ids"):
-                        try:
-                            # Use int() to test if list of integers is valid.
-                            submissions_ids = payload.get("submissions_ids", [])
-                            or_ = {u"$or": [{u"_id": int(submission_id)} for submission_id in submissions_ids]}
-                            query.update(or_)
-                        except ValueError:
-                            raise ParseError(_("Invalid submissions ids: %(submissions_ids)s"
-                                               % {'submissions_ids': json.dumps(payload.get("submissions_ids"))}))
-
-                        filter_.update({"id__in": submissions_ids})
-                    # Third scenario / Modify all submissions in form, but confirmation param must be among payload
-                    elif payload.get("confirm", False) is not True:
-                        http_status = status.HTTP_400_BAD_REQUEST
-                        response = {"detail": _("No confirmations provided")}
+        # Second scenario / Get submissions based on list of ids
+        elif payload.get('submissions_ids'):
+            try:
+                # Use int() to test if list of integers is valid.
+                instances_ids = [int(submission_id)
+                                 for submission_id in payload.get(
+                                    'submissions_ids', [])]
```

Comment:
At this point we know `payload['submissions_ids']` exists; there's no need for a default. Again, I'd suggest `try`ing to get `payload['submissions_ids']`, handling the `KeyError`, and using the `else` block for the logic that should be invoked when submissions' IDs were specified.

Reply:
Let's discuss about this.